metadata/reportssql/hiv_patient_level_report_functions.sql

```diff
@@ -0,0 +1,335 @@
+    ELSEIF (relationship = "RELATIONSHIP_BIO_MOTHER" OR relationship = "RELATIONSHIP_BIO_FATHER") THEN
+        RETURN "Biological parent of an index child";
+    ELSEIF (relationship = "RELATIONSHIP_SIBLING") THEN
+        RETURN "Sibling of index case <15 years";
+    ELSE
+        RETURN "Other";
+    END IF;
+END$$
+DELIMITER ;
+
+-- getRadetDefaulterNotificationMethod
+
+DROP FUNCTION IF EXISTS getRadetDefaulterNotificationMethod;
+
+DELIMITER $$
+CREATE FUNCTION getRadetDefaulterNotificationMethod(
+    p_patientId INT(11)) RETURNS VARCHAR(255)
+    DETERMINISTIC
+BEGIN
+    DECLARE method VARCHAR(50) DEFAULT getPatientMostRecentProgramAttributeCodedValueFromName(p_patientId, 'PROGRAM_MANAGEMENT_1_NOTIFICATION_METHOD', 'en');
+
+    IF (method = "By index") THEN
+        RETURN "Index";
+    ELSEIF (method = "By service provider") THEN
+        RETURN "Service provider";
+    ELSEIF (method = "Contract") THEN
+        RETURN method;
+    ELSEIF (method = "Dual Notification") THEN
+        RETURN method;
+    ELSEIF (method = "Security risk (not recommended)") THEN
+        RETURN method;
+    ELSE
+        RETURN "";
+    END IF;
+END$$
+DELIMITER ;
+
+-- getRadetDefaulterNotificationOutcome
+
+DROP FUNCTION IF EXISTS getRadetDefaulterNotificationOutcome;
+
+DELIMITER $$
+CREATE FUNCTION getRadetDefaulterNotificationOutcome(
+    p_patientId INT(11)) RETURNS VARCHAR(255)
+    DETERMINISTIC
+BEGIN
+    DECLARE outcome VARCHAR(50) DEFAULT getPatientMostRecentProgramAttributeCodedValueFromName(p_patientId, 'PROGRAM_MANAGEMENT_3_NOTIFICATION_OUTCOME', 'en');
+
+    IF (outcome = "Refused Testing") THEN
+        RETURN "RT= Refused Testing";
+    ELSEIF (outcome = "Accepted Testing") THEN
+        RETURN "AT  =  Accepted Testing";
+    ELSEIF (outcome = "Positive, on TX") THEN
+        RETURN "PT= Positive, on Tx";
+    ELSEIF (outcome = "Positive, NOT on TX") THEN
+        RETURN "PXT=  Positive, NOT on TX";
+    ELSE
+        RETURN "";
+    END IF;
+END$$
+DELIMITER ;
+
+-- getRadetTestedLocation
+
+DROP FUNCTION IF EXISTS getRadetTestedLocation;
+
+DELIMITER $$
+CREATE FUNCTION getRadetTestedLocation(
+    p_patientId INT(11)) RETURNS VARCHAR(255)
+    DETERMINISTIC
+BEGIN
+    DECLARE testedLocation VARCHAR(50) DEFAULT getTestedLocation(p_patientId);
+
+    IF (testedLocation LIKE "Community%") THEN
+        RETURN "Community/Communauté";
+    ELSE RETURN "Facility/FOSA";
+    END IF;
+
+END$$
+DELIMITER ;
+
+-- getRadetIndexType
+
+DROP FUNCTION IF EXISTS getRadetIndexType;
+
+DELIMITER $$
+CREATE FUNCTION getRadetIndexType(
+    p_patientId INT(11)) RETURNS VARCHAR(255)
+    DETERMINISTIC
+BEGIN
+    DECLARE daysBetweenHIVPosAndART INT(11) DEFAULT getDaysBetweenHIVPosAndART(p_patientId);
+    DECLARE viralLoadResult INT(11) DEFAULT getViralLoadTestResult(p_patientId);
+
+    IF (daysBetweenHIVPosAndART > 0 AND daysBetweenHIVPosAndART <= 30) THEN
+        RETURN 'Index Case New HTS POS and initiated on treatment within a month';
```

Comment:
@lauravignoli The above option reads "Index Case New HTS POS and initiated on treatment **this month**" in the radet report